### PR TITLE
Add MarmoymError and ErrorType

### PR DESCRIPTION
**Pull request by enginehenryed**
#29
MarmoymError is an object that represents errors defined
in Marmoym. ErrorType is predefined types of MarmoymError.

This commit replaces error throwings within UserSignInController.

### CODE REVIEW

File: src/routes/v1/testRoute.ts [hunk 1]
```diff
@@ -36,4 +37,11 @@ router.get('/', tokenAuthHandler, async (req: Request, res: Response) => {
   
 });
 
+import ErrorType from '../../constants/ErrorType';
+
+router.get("/a", (req, res) => {
+  throw new MarmoymError(ErrorType.USER_INCORRECT_PASSWORD);
+  // respond(res, "success");
```
Comment (jcooky) [inline src/routes/v1/testRoute.ts]: 이거 주석은 일부러 남겨놓으신건가요?

**Comment by enginehenryed** [comment]
Reason that this commit did not get passed the travis => no travis.yml file

File: src/routes/v1/testRoute.ts (hunk 1 above)
Comment (enginehenryed) [inline src/routes/v1/testRoute.ts]: 아니요 처리후 다시 notify할게요

**Comment by enginehenryed** [comment]
@jcooky Your comment was deleted by forced-pushing. But the ones you suggested were all done in this commit, I believe. Please check this out and let me know if anything is unclear.

File: src/constants/ErrorType/UserError.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default {
+  "USER_INCORRECT_PASSWORD": {
+    code: 401001,
+    msg: "text"
+  },
+  "USER_NOT_FOUND": {
+    code: 401000,
+    msg: "text"
```
Comment (jcooky) [inline src/constants/ErrorType/UserError.ts]: Message를 제대로 적던가 아니면 TODO 주석을 달아서 수정만 해주시면 될거같애요.

Comment (enginehenryed) [inline src/constants/ErrorType/UserError.ts]: Modified

**Pull request merged**